src/Compilers/CSharp/Portable/Syntax/SyntaxKindFacts.cs

```diff
@@ -1046,11 +1047,28 @@ public static SyntaxKind GetPreprocessorKeywordKind(string text)
             }
         }
 
+        private static SyntaxKind[] _lazyContextualKeywordKindsArray;
```

Comment:
Yes.  it would be an immutable array under hte covers.  I'm just saying it can be constructed with an ArrayBuilder so there isn't the random List allocation/free.